packages/types/src/authentication/common/provider.ts

```diff
@@ -2,4 +2,21 @@ export type AuthenticationResponse = {
   success: boolean
   authUser?: any
   error?: string
+  location?: string
+}
+
+export type AuthModuleProviderConfig = {
+  name: string
+  scopes: Record<string, AuthProviderScope>
+}
+
+export type AuthProviderScope = { domain?: string } & Record<string, string>
```

Comment:
nit(non-blocking): Should this type be entirely open-ended? Are we using `domain` yet?
```suggestion
export type AuthProviderScope = Record<string, unknown>
```